packages/circuits/src/ReportIdentityProof.ts

```diff
@@ -0,0 +1,22 @@
+import { BaseProof, Prover } from '@unirep/circuits'
+import { Groth16Proof } from 'snarkjs'
+
+export class ReportIdentityProof extends BaseProof {
+    public reportNullifier: bigint
+    public attesterId: bigint
+    public epoch: bigint
+    public stateTreeRoot: bigint
+
+    constructor(
+        publicSignals: (bigint | string)[],
+        proof: Groth16Proof,
+        prover?: Prover
+    ) {
+        super(publicSignals, proof, prover)
+        this.reportNullifier = this.publicSignals[0]
+        this.attesterId = this.publicSignals[1]
+        this.epoch = this.publicSignals[2]
+        this.stateTreeRoot = this.publicSignals[3]
+        ;(this as any).circuit = 'reportIdentityProof'
```

Comment:
I am quite curious about why can't it be `this.circuit = 'reportIdentityProof'` (just curious about it 👍 )

Reply:
fixed

Reply:
The type of the instance variable `circuit` which is inherited from `BaseProof` class is `Circuit` type in Unirep SDK.

Here is Circuit type: https://github.com/Unirep/Unirep/blob/e57834c1244e76a1e3e8289b0f92b072c2fc8fb8/packages/circuits/src/type.ts#L57

As you can see, 'reportIdentityProof is not included in Circuit. So here we have to cast this instance to any to ignore type-checking error.

Maybe a better solution is to refactor BaseProof with generic type. We can fork this class or make a pull request to Unirep Repository. I think this task can be left to @Lisooo790926.